# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/powerpc64-2.patch
# Copyright (C) 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

With leading dots I get unresolved symbols ...

  - Rene Rebe <rene@exactcode.de>

diff -ur dietlibc-0.29/Makefile dietlibc-0.29-fixed/Makefile
--- dietlibc-0.29/Makefile	2005-05-18 16:38:55.000000000 +0200
+++ dietlibc-0.29-fixed/Makefile	2005-12-08 10:41:39.000000000 +0100
@@ -158,11 +158,11 @@
 
 $(OBJDIR)/pthread_%.o: libpthread/pthread_%.c
 	$(CROSS)$(CC) -I. -isystem include $(CFLAGS) -c $< -o $@
-	$(COMMENT) -$(CROSS)strip -x -R .comment -R .note $@
+	$(COMMENT) -$(STRIP) -x -R .comment -R .note $@
 
 $(OBJDIR)/%.o: %.c
 	$(CROSS)$(CC) -I. -isystem include $(CFLAGS) -c $< -o $@
-	$(COMMENT) -$(CROSS)strip -x -R .comment -R .note $@
+	$(COMMENT) -$(STRIP) -x -R .comment -R .note $@
 
 ifeq ($(shell $(CC) -v 2>&1 | grep "gcc version"),gcc version 4.0.0)
 SAFE_CFLAGS=$(shell echo $(CFLAGS)|sed 's/-Os/-O2/')
@@ -182,45 +182,45 @@
 $(OBJDIR)/clone.o
 
 $(OBJDIR)/dietlibc.a: $(DIETLIBC_OBJ) $(OBJDIR)/start.o
-	$(CROSS)ar cru $@ $(DIETLIBC_OBJ)
+	$(AR) cru $@ $(DIETLIBC_OBJ)
 
 $(OBJDIR)/librpc.a: $(LIBRPCOBJ)
-	$(CROSS)ar cru $@ $(LIBRPCOBJ)
+	$(AR) cru $@ $(LIBRPCOBJ)
 
 $(OBJDIR)/libcrypt.a:
 	touch dummy.c
 	$(CROSS)$(CC) -c dummy.c
-	$(CROSS)ar cru $@ dummy.o
+	$(AR) cru $@ dummy.o
 	rm -f dummy.c dummy.o
 
 dummy.o:
 
 LIBLATIN1_OBJS=$(patsubst liblatin1/%.c,$(OBJDIR)/%.o,$(wildcard liblatin1/*.c))
 $(OBJDIR)/liblatin1.a: $(LIBLATIN1_OBJS)
-	$(CROSS)ar cru $@ $^
+	$(AR) cru $@ $^
 
 $(OBJDIR)/libgmon.a: $(LIBGMON_OBJS)
-	$(CROSS)ar cru $@ $^
+	$(AR) cru $@ $^
 
 $(OBJDIR)/libpthread.a: $(LIBPTHREAD_OBJS) dietfeatures.h
-	$(CROSS)ar cru $@ $(LIBPTHREAD_OBJS)
+	$(AR) cru $@ $(LIBPTHREAD_OBJS)
 
 $(OBJDIR)/libcompat.a: $(LIBCOMPATOBJ)
-	$(CROSS)ar cru $@ $(LIBCOMPATOBJ)
+	$(AR) cru $@ $(LIBCOMPATOBJ)
 
 $(OBJDIR)/libm.a: $(LIBMATHOBJ)
-	$(CROSS)ar cru $@ $(LIBMATHOBJ)
+	$(AR) cru $@ $(LIBMATHOBJ)
 
 LD_UNSET = env -u LD_RUN_PATH
 
 $(OBJDIR)/libdietc.so: $(OBJDIR)/dietlibc.a
-	$(LD_UNSET) $(CROSS)ld -whole-archive -shared -o $@ $^
+	$(LD_UNSET) $(LD) -whole-archive -shared -o $@ $^
 
 dyn: dyn_lib
 
 # added dynamic linker
 $(OBJDIR)/libdl.a: $(LIBDLOBJ)
-	$(CROSS)ar cru $@ $(LIBDLOBJ)
+	$(AR) cru $@ $(LIBDLOBJ)
 
 dyn_lib: $(PICODIR) $(PICODIR)/libc.so $(PICODIR)/dstart.o \
 	$(PICODIR)/dyn_so_start.o $(PICODIR)/dyn_start.o $(PICODIR)/dyn_stop.o \
@@ -232,18 +232,18 @@
 
 $(PICODIR)/pthread_%.o: libpthread/pthread_%.c
 	$(CROSS)$(CC) -I. -isystem include $(CFLAGS) -fPIC -D__DYN_LIB -c $< -o $@
-	$(COMMENT) $(CROSS)strip -x -R .comment -R .note $@
+	$(COMMENT) $(STRIP) -x -R .comment -R .note $@
 
 $(PICODIR)/%.o: %.c
 	$(CROSS)$(CC) -I. -isystem include $(CFLAGS) -fPIC -D__DYN_LIB -c $< -o $@
-	$(COMMENT) $(CROSS)strip -x -R .comment -R .note $@
+	$(COMMENT) $(STRIP) -x -R .comment -R .note $@
 
 $(PICODIR)/dstart.o: start.S
 	$(CROSS)$(CC) -I. -isystem include $(CFLAGS) -fPIC -D__DYN_LIB -c $< -o $@
 
 $(PICODIR)/dyn_so_start.o: dyn_start.c
 	$(CROSS)$(CC) -I. -isystem include $(CFLAGS) -fPIC -D__DYN_LIB -D__DYN_LIB_SHARED -c $< -o $@
-	$(COMMENT) $(CROSS)strip -x -R .comment -R .note $@
+	$(COMMENT) $(STRIP) -x -R .comment -R .note $@
 
 DYN_LIBC_PIC = $(LIBOBJ) $(LIBSTDIOOBJ) $(LIBUGLYOBJ) \
 $(LIBCRUFTOBJ) $(LIBCRYPTOBJ) $(LIBSHELLOBJ) $(LIBREGEXOBJ)
@@ -292,19 +292,19 @@
 
 $(OBJDIR)/diet: $(OBJDIR)/start.o $(OBJDIR)/dyn_start.o diet.c $(OBJDIR)/dietlibc.a $(OBJDIR)/dyn_stop.o
 	$(CROSS)$(CC) -isystem include $(CFLAGS) -nostdlib -o $@ $^ -DDIETHOME=\"$(HOME)\" -DVERSION=\"$(VERSION)\" -lgcc
-	$(CROSS)strip -R .comment -R .note $@
+	$(STRIP) -R .comment -R .note $@
 
 $(OBJDIR)/diet-i: $(OBJDIR)/start.o $(OBJDIR)/dyn_start.o diet.c $(OBJDIR)/dietlibc.a $(OBJDIR)/dyn_stop.o
 	$(CROSS)$(CC) -isystem include $(CFLAGS) -nostdlib -o $@ $^ -DDIETHOME=\"$(prefix)\" -DVERSION=\"$(VERSION)\" -DINSTALLVERSION -lgcc
-	$(CROSS)strip -R .comment -R .note $@
+	$(STRIP) -R .comment -R .note $@
 
 $(PICODIR)/diet-dyn: $(PICODIR)/start.o $(PICODIR)/dyn_start.o diet.c
 	$(LD_UNSET) $(CROSS)$(CC) -isystem include $(CFLAGS) -fPIC -nostdlib -o $@ $^ -DDIETHOME=\"$(HOME)\" -D__DYN_LIB -DVERSION=\"$(VERSION)\" -L$(PICODIR) -lc -lgcc $(PICODIR)/dyn_stop.o -Wl,-dynamic-linker=$(HOME)/$(PICODIR)/libdl.so
-	$(CROSS)strip -R .command -R .note $@
+	$(STRIP) -R .command -R .note $@
 
 $(PICODIR)/diet-dyn-i: $(PICODIR)/start.o $(PICODIR)/dyn_start.o diet.c
 	$(LD_UNSET) $(CROSS)$(CC) -isystem include $(CFLAGS) -fPIC -nostdlib -o $@ $^ -DDIETHOME=\"$(prefix)\" -D__DYN_LIB -DVERSION=\"$(VERSION)\" -L$(PICODIR) -lc -lgcc $(PICODIR)/dyn_stop.o -Wl,-dynamic-linker=$(ILIBDIR)/libdl.so -DINSTALLVERSION
-	$(CROSS)strip -R .command -R .note $@
+	$(STRIP) -R .command -R .note $@
 
 $(OBJDIR)/djb: $(OBJDIR)/compile $(OBJDIR)/load
 
diff -ur dietlibc-0.29/dietfeatures.h dietlibc-0.29-fixed/dietfeatures.h
--- dietlibc-0.29/dietfeatures.h	2005-12-08 11:49:28.000000000 +0100
+++ dietlibc-0.29-fixed/dietfeatures.h	2005-12-08 10:41:39.000000000 +0100
@@ -41,7 +41,6 @@
 /* on i386, Linux has an alternate syscall method since 2002/12/16 */
 /* on my Athlon XP, it is twice as fast, but it's only in kernel 2.5 */
 /* 20040118: enabling this breaks User Mode Linux!  It's their fault. */
-#define WANT_SYSENTER
 
 #define WANT_LINKER_WARNINGS
 
diff -ur dietlibc-0.29/ppc64/__longjmp.S dietlibc-0.29-fixed/ppc64/__longjmp.S
--- dietlibc-0.29/ppc64/__longjmp.S	2005-12-08 11:49:28.000000000 +0100
+++ dietlibc-0.29-fixed/ppc64/__longjmp.S	2005-12-08 11:48:51.000000000 +0100
@@ -1,12 +1,12 @@
 #include <setjmp.h>
 
-		.quad	.__longjmp,.TOC.@tocbase,0
+		.quad	__longjmp,.TOC.@tocbase,0
 		.text
 		.size	__longjmp,24
-		.type	.__longjmp,@function
-		.globl	.__longjmp
+		.type	__longjmp,@function
+		.globl	__longjmp
 .align 2
-.__longjmp:
+__longjmp:
 	ld  1,(JB_GPR1*8)(3)
 	ld  2,(JB_GPR2*8)(3)
 	ld  0,(JB_LR*8)(3)
@@ -51,4 +51,4 @@
 	lfd 31,((JB_FPRS+17)*8)(3)
 	mr 3,4
 	blr
-.size .__longjmp,.-.__longjmp
+.size __longjmp,.-__longjmp
diff -ur dietlibc-0.29/ppc64/setjmp.S dietlibc-0.29-fixed/ppc64/setjmp.S
--- dietlibc-0.29/ppc64/setjmp.S	2005-12-08 11:49:28.000000000 +0100
+++ dietlibc-0.29-fixed/ppc64/setjmp.S	2005-12-08 11:52:14.000000000 +0100
@@ -6,13 +6,13 @@
 #define JUMPTARGET(name) .name
 #endif
 
-		.quad	.__sigsetjmp,.TOC.@tocbase,0
+		.quad	__sigsetjmp,.TOC.@tocbase,0
 		.text
 		.size	__sigsetjmp,24
-		.type	.__sigsetjmp,@function
-		.globl	.__sigsetjmp
+		.type	__sigsetjmp,@function
+		.globl	__sigsetjmp
 .align 2
-.__sigsetjmp:
+__sigsetjmp:
         std  1,(JB_GPR1*8)(3)
         mflr 0
         std  2,(JB_GPR2*8)(3)
@@ -56,4 +56,4 @@
         std  31,((JB_GPRS+17)*8)(3)
         stfd 31,((JB_FPRS+17)*8)(3)
         b JUMPTARGET (__sigjmp_save)
-.size .__sigsetjmp,.-.__sigsetjmp
+.size __sigsetjmp,.-__sigsetjmp
